engine/src/main/scala/cromwell/engine/language/LanguageConfiguration.scala

```diff
@@ -0,0 +1,50 @@
+package cromwell.engine.language
+
+import java.util.Map.Entry
+
+import com.typesafe.config.ConfigFactory
+import cromwell.engine.language.CromwellLanguages.{CromwellLanguageName, CromwellLanguageVersion}
+
+import scala.collection.JavaConverters._
+
+case class LanguageConfigurationEntry(name: CromwellLanguageName, versions: Map[CromwellLanguageVersion, String])
+
+
+object LanguageConfiguration {
+  private val LanguagesConfig = ConfigFactory.load.getConfig("languages")
+  private val LanguageNames: Set[String] = LanguagesConfig.entrySet().asScala.map(findFirstKey).toSet
+
+  val AllLanguageEntries: List[LanguageConfigurationEntry] = LanguageNames.toList map { languageName =>
+
+    val languageConfig = LanguagesConfig.getConfig(languageName)
+    val versionSet = languageConfig.getConfig("versions")
+    val languageVersionNames: Set[String] = versionSet.entrySet().asScala.map(findFirstKey).toSet
+
+    val versions = (languageVersionNames.toList map { languageVersionName =>
+      val entry = versionSet.getConfig(s""""$languageVersionName"""")
+      languageVersionName -> entry.getString("language-factory")
+    }).toMap
+
+    LanguageConfigurationEntry(languageName, versions)
+  }
+
+  // Gets the first key in a hocon key entry (which might contain several keys, perhaps surrounded by quotes)
+  // EG
+  // foo {
+  //   "1.0" {
+  //     "x.y".z = "hello"
+  //   }
+  // }
+  // Called on the entry on line 1, returns 'foo'
+  // Called on the entry on line 2, returns '1.0'
+  // Called on the entry on line 3, returns 'x.y'
+  private def findFirstKey(entry: Entry[String, _]): String = {
+    val NoQuoteFirstKey = """([^.]*)\.(.*)""".r
+    val SingleQuotedFirstKey = """"(.*)"\.(.*)""".r
+
+    entry.getKey match {
+      case SingleQuotedFirstKey(firstKey, _) => firstKey
+      case NoQuoteFirstKey(firstKey, _) => firstKey
+    }
+  }
```

Comment:
couldn't this use a syntax that stays within HOCON?

Reply:
So HOCON does support escaped keys, and if we always write keys on separate lines there's no issue. The problem comes when you try and put two keys on the same line, like
```
x {
  y.z = foo
}
```



`BackendConfigurationEntry` solves the same problem with a different string manipulation (the `.map(_.getKey.split("\\.")` which it hides by putting it inline). That doesn't consider the possibility that the first key is quoted.